src/EventRegistry.php

```diff
@@ -57,6 +64,35 @@ class EventRegistry
     public function __construct(EventDispatcherInterface $eventDispatcher)
     {
         $this->eventDispatcher = $eventDispatcher;
+
+        // Check if the new event classes exist, if so replace the old one with the new.
+        if (class_exists(ControllerArgumentsEvent::class)) {
+            self::$eventsMap['kernel.controller_arguments'] = ControllerArgumentsEvent::class;
```

Comment:
In the new system (with these new event classes), the event class and the event name are the same! So, you'd actually need to `unset()` the original key, then add a new item to the array like this:

`ControllerArgumentsEvent::class => ControllerArgumentsEvent::class`.

I might even create a new static property with a map of the old event names to the new event class, so that you could just loop in here:

```php
private static $newEventsMap = [
    'kernel.controller_arguments' => ControllerArgumentsEvent::class
];

// .. and then in the constructor
foreach (self::$newEventsMap as $oldEventName => $newEventClass) {
    // unset old key
    // add new key
}
```

Nice solution though - it's a surprisingly tricky thing to get right.

Reply:
Thank you for the feedback ! I updated the PR with the solution you recommended. I was first thinking of using the new events by default (because in the future they will be the default ones) but I didn't find an easy solution to do this. I prefer to do oldX / X rather then X / newX because in the future you would just need to delete the oldX in order to update the code.